src/error.rs

```diff
@@ -206,17 +224,26 @@ impl<Src, Dst: ?Sized> fmt::Debug for SizeError<Src, Dst> {
 /// Produces a human-readable error message.
 impl<Src, Dst: ?Sized> fmt::Display for SizeError<Src, Dst>
 where
-    Src: Deref,
```

Comment:
I think we should keep this bound, too. Once we have `size_of_val`, we can use this bound to provide information about the length of the source.

Reply:
What do you mean "once we have `size_of_val`"? [We already do](https://doc.rust-lang.org/std/mem/fn.size_of_val.html).

(Same concern about leaking size, btw.)

Reply:
Hmmm yeah I could see that. Maybe a "rich-error-text" feature or something? We could probably gate a lot behind that so we're super anal without it but pretty verbose with it.